webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SecretApi.scala

```diff
@@ -0,0 +1,165 @@
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+package com.normation.rudder.rest.lift
+
+import com.normation.box._
+import com.normation.rudder.domain.secrets.Secret
+import com.normation.rudder.rest.ApiPath
+import com.normation.rudder.rest.ApiVersion
+import com.normation.rudder.rest.AuthzToken
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.rest.RestUtils
+import com.normation.rudder.rest.{SecretApi => API}
+import com.normation.rudder.web.services.SecretService
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import net.liftweb.json.JArray
+import net.liftweb.json.JString
+
+
+class SecretApi(
+    restExtractorService : RestExtractorService
+  , secretService        : SecretService
+) extends LiftApiModuleProvider[API] {
+
+  def schemas = API
+
+  def getLiftEndpoints(): List[LiftApiModule] = {
+    API.endpoints.map(
+      e => e match {
+        case API.GetAllSecret => GetAllSecret
+        case API.GetSecret    => GetSecret
+        case API.AddSecret    => AddSecret
+        case API.UpdateSecret => UpdateSecret
+        case API.DeleteSecret => DeleteSecret
+      }
+    )
+  }
+
+  object GetAllSecret extends LiftApiModule0 {
+    val schema        = API.GetAllSecret
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action   = schema.name
+
+      val res = for {
+        secrets <- secretService.getAllSecret.toBox
+      } yield {
+        JArray(secrets.map(Secret.serializeSecret))
+      }
+
+      RestUtils.response(restExtractor, "secrets", None)(res, req, "Error when trying to get all secrets")
+    }
+  }
+
+  object GetSecret extends LiftApiModule {
+    val schema        = API.GetSecret
+    val restExtractor = restExtractorService
+
+    def process(version: ApiVersion, path: ApiPath, id: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action   = schema.name
+
+      val res = for {
+        secret <- secretService.getSecretById(id).notOptional(s"Could not find secret with `${id}` name").toBox
+      } yield {
+        Secret.serializeSecret(secret)
+      }
+
+      RestUtils.response(restExtractor, "secrets", None)(res, req, s"Error when trying to get secret `${id}` value")
```

Comment:
`secrets` -> "secret

Reply:
secrets is good here, we always send arrays and there for the container is always plura

Reply:
(and so you need to send an array)